frontend/src/data/utils/DatasetTimeUtils.test.ts

```diff
@@ -206,3 +208,15 @@ describe('Tests getPrettyDate() function', () => {
     expect(getPrettyDate('ABCD-YZ')).toEqual('ABCD-YZ')
   })
 })
+
+describe('Tests getMostRecentYearAsString()', () => {
+test('correct year string is returned', async() => {
+  const mockDataFrame = new DataFrame([
+    { time_period: '2019', pct_share_of_women_state_leg: 10 },
+    { time_period: '2017', pct_share_of_women_us_congress: 20 },
+])
+
+const mostRecentYear = getMostRecentYearAsString(mockDataFrame, ['pct_share_of_women_us_congress'])
+expect(mostRecentYear).toEqual('2017')
+
```

Comment:
this is great. can you expand the test df a little and add a few more years for each metric_id? the test you added shows that it detects the most recent year for the given metricId which is great, i'd love to also see:
- that it still gives '2017' for pct_share_of_women_us_congress when there is a '2016' year
- that it gives '2019' for 'pct_share_of_women_state_leg' 
- that we handle the case where the requested metric id isn't in the df at all? what happens there? 
-

Reply:
Absolutely!